src/sys/unix/kqueue.rs

```diff
@@ -74,7 +74,11 @@ impl Selector {
     ) -> io::Result<bool> {
         let timeout = timeout.map(|to| libc::timespec {
             tv_sec: cmp::min(to.as_secs(), time_t::max_value() as u64) as time_t,
-            tv_nsec: libc::c_long::from(to.subsec_nanos()),
+            // `Duration::subsec_nanos` is guaranteed to be less than one
+            // billion (the number of nanoseconds in a second), making the
+            // cast to i32 safe. The cast itself is needed for platforms
+            // where C's long is only 32 bits.
```

Comment:
The comment isn't quite right.  The cast is only needed for platforms where C's long is `i32`.  No cast is needed where long is `u32`.

Reply:
Isn't C `long` always signed? According to https://en.wikipedia.org/wiki/C_data_types: "Long signed integer type. Capable of containing at least the [−2,147,483,647, +2,147,483,647] range;[3][4] thus, it is at least 32 bits in size."

Reply:
Yes it is always signed, except when I'm being stupid.